### PR TITLE
[testharness.js] Remove deprecated API

### PR DESCRIPTION
**Pull request by jugglinmike**
The testharness.js functions `assert_exists` and `assert_not_exists`
have been labeled "deprecated" since 2012-01-31 [1]. Remove the
functions and replace all existing references with equivalent
assertions.

[1] 1780607fc4582b65b7fcdded3e39e79e95c8d915

### CODE REVIEW

File: fetch/corb/script-html-via-cross-origin-blob-url.sub.html [hunk 1]
```diff
@@ -19,7 +19,7 @@
   }
 
   function step2_processSubframeMsg(msg) {
-    assert_not_exists(msg, 'error');
+    assert_false(msg.haOwnProperty('error'), 'unexpected property found: "error"');
```
Comment (jgraham) [inline fetch/corb/script-html-via-cross-origin-blob-url.sub.html]: `haOwnProperty`

**Review by jgraham — CHANGES_REQUESTED** [review]

File: fetch/corb/script-html-via-cross-origin-blob-url.sub.html (hunk 1 above)
Comment (jugglinmike) [reply fetch/corb/script-html-via-cross-origin-blob-url.sub.html]: Oof, yes. Also, I introduced a syntax error in testharness.js. Both are fixed now

**Comment by jugglinmike** [comment]
To validate the current revision, I collected the names of the effected tests:

    $ ./wpt tests-affected master
    INFO:manifest:Updating manifest
    eventsource/eventsource-prototype.htm
    fetch/corb/script-html-via-cross-origin-blob-url.sub.html
    html/browsers/history/the-history-interface/001.html
    html/browsers/history/the-history-interface/002.html
    html/semantics/embedded-content/media-elements/track/track-element/track-change-event.html
    webrtc/getstats.html
    xhr/responseXML-unavailable-in-worker.html

And executed them in `master` and on this branch using Chrome:

    $ echo "eventsource/eventsource-prototype.htm
    fetch/corb/script-html-via-cross-origin-blob-url.sub.html
    html/browsers/history/the-history-interface/001.html
    html/browsers/history/the-history-interface/002.html
    html/semantics/embedded-content/media-elements/track/track-element/track-change-event.html
    webrtc/getstats.html
    xhr/responseXML-unavailable-in-worker.html" | xargs ./wpt run chrome

<details>
  <summary>result on `master`</summary>

    web-platform-test
    ~~~~~~~~~~~~~~~~~
    Ran 90 checks (7 tests, 83 subtests)
    Expected results: 88
    Unexpected results: 2
      subtest: 2 (2 fail)

    Unexpected Results
    ------------------
    /html/browsers/history/the-history-interface/001.html
      FAIL pushState must remove any tasks queued by the history traversal task source - assert_equals: expected "test4" but got "test"
        at Test.<anonymous> (http://web-platform.test:8000/html/browsers/history/the-history-interface/001.html:143:13)
        at Test.step (http://web-platform.test:8000/resources/testharness.js:1558:25)
        at test (http://web-platform.test:8000/resources/testharness.js:544:18)
        at tests8 (http://web-platform.test:8000/html/browsers/history/the-history-interface/001.html:142:9)
    /xhr/responseXML-unavailable-in-worker.html
      FAIL XMLHttpRequest's responseXML property should not be exposed in workers. - assert_not_exists: responseXML should not be on the prototype. unexpected property "responseXML" found
        at Test.<anonymous> (http://web-platform.test:8000/xhr/resources/responseXML-unavailable-in-worker.js:6:5)
        at Test.step (http://web-platform.test:8000/resources/testharness.js:1558:25)
        at test (http://web-platform.test:8000/resources/testharness.js:544:18)
        at http://web-platform.test:8000/xhr/resources/responseXML-unavailable-in-worker.js:3:1
     0:31.45 INFO Closing logging queue
     0:31.45 INFO queue closed

</details>

<details>
  <summary>result on thi branch</summary>

    web-platform-test
    ~~~~~~~~~~~~~~~~~
    Ran 90 checks (7 tests, 83 subtests)
    Expected results: 88
    Unexpected results: 2
      subtest: 2 (2 fail)

    Unexpected Results
    ------------------
    /html/browsers/history/the-history-interface/001.html
      FAIL pushState must remove any tasks queued by the history traversal task source - assert_equals: expected "test4" but got "test"
        at Test.<anonymous> (http://web-platform.test:8000/html/browsers/history/the-history-interface/001.html:143:13)
        at Test.step (http://web-platform.test:8000/resources/testharness.js:1546:25)
        at test (http://web-platform.test:8000/resources/testharness.js:544:18)
        at tests8 (http://web-platform.test:8000/html/browsers/history/the-history-interface/001.html:142:9)
    /xhr/responseXML-unavailable-in-worker.html
      FAIL XMLHttpRequest's responseXML property should not be exposed in workers. - assert_false: responseXML should not be on the prototype. expected false got true
        at Test.<anonymous> (http://web-platform.test:8000/xhr/resources/responseXML-unavailable-in-worker.js:6:5)
        at Test.step (http://web-platform.test:8000/resources/testharness.js:1546:25)
        at test (http://web-platform.test:8000/resources/testharness.js:544:18)
        at http://web-platform.test:8000/xhr/resources/responseXML-unavailable-in-worker.js:3:1
     0:32.10 INFO Closing logging queue
     0:32.10 INFO queue closed

</details>

**Review by jgraham — APPROVED** [review]

**Review by gsnedders — APPROVED** [review]

**Comment by annevk** [comment]
Perhaps we should also have assert_not_own_property?

**Comment by jgraham** [comment]
@annevk Yes, I think that would make sense, but let's do it as a followup so this doesn't run too much risk of conflicts.

**Pull request merged**